src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -440,8 +441,26 @@ private void delete(boolean cut, List<BibEntry> entries) {
             return;
         }
 
+        // Delete selected entries
         getUndoManager().addEdit(new UndoableRemoveEntries(bibDatabaseContext.getDatabase(), entries, cut));
         bibDatabaseContext.getDatabase().removeEntries(entries);
+
+        // Not delete files when user select cut
+        if (!cut) {
+            // Keep track of linked files from selected entries
+            List<LinkedFile> linkedFileList = new ArrayList<>();
+
+            for (BibEntry entry : entries) {
+                List<LinkedFile> linkedFilesFromEntry = entry.getFiles();
+                linkedFileList.addAll(linkedFilesFromEntry);
+            }
+
+            // Delete files which linked to selected entries when user select deletion
+            if (linkedFileList.size() > 0 && showLinkedFileDeleteConfirmationDialog(linkedFileList)) {
+                new DeleteFileAction(dialogService, preferencesService, bibDatabaseContext, null, null).deleteFileFromDisk(linkedFileList);
```

Comment:
That `deleteFileFromDisk` is public is really strange. However, I see why you use that method --> nice use of `dialogService`.

Proposal: Make `deleteFileFromDisk` static, have all parameters passed inside the `DeleteFileAction` and use that static method here --> no need for a `new`...

Reply:
I am trying to change `deleteFileFromDisk` to a static method, but this causes some variables in the `DeleteFileAction` class to become static🤔. I will probably take some time to modify it carefully.